Simulator/Recorders/Hdf5Recorder.h

```diff
@@ -70,9 +70,9 @@ class Hdf5Recorder : public Recorder {
    /// Registered to OperationManager as Operation::printParameters
    virtual void printParameters() override;
 
-   virtual void registerVariable(std::string varName, EventBuffer &recordVar) override;
+   virtual void registerVariable(std::string varName, RecordableBase *recordVar) override;
 
-   virtual void registerVariable(std::string varName, vector<EventBuffer> &recordVar) override;
+   virtual void registerVariable(std::string varName, vector<RecordableBase *> recordVars) override;
```

Comment:
Methods need Doxygen comments. If you want to defer this, make an issue and respond/resolve this.